docs/tutorials/handler.md

```diff
@@ -0,0 +1,389 @@
+Rug CLI looks like.  We will add an event handler called
+"CloseIssueThanks" with the following command.
+
+```console
+$ rug edit atomist-rugs:rug-editors:AddTypeScriptEventHandler \
+    handlerName=CloseIssueThanks \
+    description="simple event handler that sends a thank you message" \
+    pathExpression="/Issue()"
+```
+
+This editor will create files for your event handler and its test.
+The path expression we provide is the piece of information that
+matches incoming events to event handlers.  In this example, we are
+matching against all events having to do with issues.
+
+Rugs are most often written in [TypeScript][ts], which is an superset
+of JavaScript that introduces strong typing and transpiles into
+JavaScript.  To fully benefit from the TypeScript development process,
+you should install [Node.js][node], which also installs the `npm`
+utility.  Once `npm` is available on your system, you can install
+TypeScript tools and the [yarn][] package manager with the following
+command.
+
+```console
+$ npm install -g yarn typescript tslint
+```
+
+Once you have those tools installed, change into the `.atomist`
+directory of your Rug project and install the project's TypeScript
+dependencies using yarn.
+
+```console
+$ cd .atomist
+$ yarn install
+```
+
+<div class="ss-container">
+  <img src="../images/yarn-install.png" alt="Install TypeScript Dependencies with yarn" class="ss-small">
+</div>
+
+[ts]: https://www.typescriptlang.org/ (TypeScript)
+[node]: https://nodejs.org/ (Node.js)
+[yarn]: https://yarnpkg.com/en/ (Yarn Package Manager)
+
+!!! tip "TypeScript Development"
+    If you do not have a TypeScript development environment set up, we
+    recommend using [Visual Studio Code][vs-code].
+
+[vs-code]: https://code.visualstudio.com/ (Visual Studio Code)
+
+With our tooling and dependencies installed, we can begin developing
+our Rug event handler.  Following good test-driven development (TDD)
+practices, we will first update our tests to check for the proper
+behavior.  You can write unit tests as you typically would when
+developing TypeScript or JavaScript, for example using [Mocha][mocha].
+In addition, Rug development
+supports [behavior-driven development (BDD)][bdd] testing
+using [Gherkin][gherkin].  In this example we will focus on BDD
+testing.
+
+[mocha]: https://mochajs.org/ (Mocha Testing Framework)
+[bdd]: https://en.wikipedia.org/wiki/Behavior-driven_development (Behavior-Driven Development)
+[gherkin]: https://cucumber.io/docs/reference (Gherkin Testing)
+
+Open the sample feature file created by the editor we ran above in
+your preferred editor.  It will be located at
+`.atomist/tests/handlers/event/CloseIssueThanksTest.feature`.  Edit
+the file to have the following contents and save it.
+
+```gherkin
+{!../../.atomist/tests/handlers/event/CloseIssueThanksTest.feature!}
+```
+
+Here we describe the general feature and provide two scenarios.  The
```

Comment:
Should we emphasize `feature` and `scenario` since they are Gherkin/BDD concepts?